hs/src/Reach/Connector/ETH_Solidity.hs

```diff
@@ -1654,6 +1661,10 @@ solPLProg PLProg {plp_opts = plo, plp_init = dli, plp_cpprog = CPProg { cpp_at =
           let vk = pretty $ vk_
           let (dom, rng) = itype2arr t
           args <- mapM (allocDLVar at) dom
+          when (length args > apiMaxArgs) $
+            error $ "The ETH connector supports Views that have up to " <>
+            show apiMaxArgs <> " arguments, but " <> vk_ <> " uses " <>
+            show (length args) <> "."
```

Comment:
Define something like

https://github.com/reach-sh/reach-lang/blob/master/hs/src/Reach/Connector/ALGO.hs#L55-L71

Probably there's just one constructor `Err_SolTooManyArgs String String Int` and this is

```
expect_throw Nothing at $ Err_SolTooManyArgs "Views" vk_ (length args)
```

https://github.com/reach-sh/reach-lang/blob/5124cfb30e147afe07c9a8a1cb8fdab74b3606a0/hs/src/Reach/AST/Base.hs#L214

And you have to add entries to the error code index --- https://github.com/reach-sh/reach-lang/blob/057eca2aba25a4d398f905fae4ab4a5611a54873/docs/src/rsh/errors/index.md

And a test case with the same name

Reply:
Ok, thanks.  I had started writing stuff like in the ALGO connector, but then I looked at how those errors were used and instead of `expect_throw` it used some other stuff defined in that file, so I was under the impression that `expect_throw` wasn't available and something else was needed.